src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -116,17 +147,90 @@ void ColumnIsNullTableScanImpl::_scan_value_segment(const BaseValueSegment& segm
 
   auto iterable = NullValueVectorIterable{segment.null_values()};
 
-  const auto invert = _predicate_condition == PredicateCondition::IsNotNull;
+  const auto invert = predicate_condition == PredicateCondition::IsNotNull;
+  const auto functor = [&](const auto& value) {
+    return invert ^ value.is_null();
+  };
+  iterable.with_iterators(position_filter, [&](auto iter, auto end) {
+    _scan_with_iterators<false>(functor, iter, end, chunk_id, matches);
+  });
+}
+
+void ColumnIsNullTableScanImpl::_scan_dictionary_segment(
+    const BaseDictionarySegment& segment, const ChunkID chunk_id, RowIDPosList& matches,
+    const std::shared_ptr<const AbstractPosList>& position_filter) {
+  if (_matches_all(segment)) {
+    _add_all(chunk_id, matches, position_filter ? position_filter->size() : segment.size());
+    return;
+  }
+
+  if (_matches_none(segment)) {
+    ++num_chunks_with_early_out;
+    return;
+  }
+
+  auto iterable = create_iterable_from_attribute_vector(segment);
+
+  const auto invert = predicate_condition == PredicateCondition::IsNotNull;
+  const auto functor = [&](const auto& value) {
+    return invert ^ value.is_null();
+  };
+
+  iterable.with_iterators(position_filter, [&](auto iter, auto end) {
+    _scan_with_iterators<false>(functor, iter, end, chunk_id, matches);
+  });
+}
+
+template <typename T>
+void ColumnIsNullTableScanImpl::_scan_run_length_segment(
+    const RunLengthSegment<T>& segment, const ChunkID chunk_id, RowIDPosList& matches,
+    const std::shared_ptr<const AbstractPosList>& position_filter) {
+  auto iterable = NullValueVectorIterable{*segment.null_values()};
+
+  const auto invert = predicate_condition == PredicateCondition::IsNotNull;
+  const auto functor = [&](const auto& value) {
+    return invert ^ value.is_null();
+  };
+
+  iterable.with_iterators(position_filter, [&](auto iter, auto end) {
+    _scan_with_iterators<false>(functor, iter, end, chunk_id, matches);
+  });
+}
+
+template <typename T>
+void ColumnIsNullTableScanImpl::_scan_LZ4_segment(const LZ4Segment<T>& segment, const ChunkID chunk_id,
+                                                  RowIDPosList& matches,
+                                                  const std::shared_ptr<const AbstractPosList>& position_filter) {
+  auto iterable = NullValueVectorIterable{*segment.null_values()};
+
+  const auto invert = predicate_condition == PredicateCondition::IsNotNull;
```

Comment:
Lot's of code duplication here. Can we lower this somehow?

Is it only the first parameter that differs?

Reply:
I would also vote for sth like `_scan_null_value_vector` that takes the null vector as input.